### PR TITLE
Upgrade libraries to latest and migrated code

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

* Upgraded all libraries to latest version, explicitly setting to the minor version when available.
* Migrated `hyper` web server code to latest version.
* Added a VSCode devcontainer for a Rust dev environment.

This should invalidate a few Renovate PRs which simply upgrade the libraries without proper code migration.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,67 +1,96 @@
...
+use std::net::IpAddr;
+use tokio::net::TcpListener;
+
+async fn location(
+    req: Request<hyper::body::Incoming>
+) -> Result<Response<Full<Bytes>>, Infallible> {
+    let uri = req.uri().to_string();
+
+    let path = env::var("MMDB_PATH").unwrap_or("./GeoLite2-City.mmdb".to_string());
+    let reader = maxminddb::Reader::open_readfile(path).unwrap();
```
Comment (jimleroyer) [inline src/main.rs:27]: I realized AFK: I need to move this file reading outside of the request processing and at server' startup.

**Review by ben851 — APPROVED** [review]
I like curly brackets

**Review by sastels — APPROVED** [review]
I like dev containers.

**Review by sylviamclaughlin — APPROVED** [review]
Looks good Jimmy! 🎉

**Pull request merged**